crates/rolldown/src/utils/chunk/finalize_chunks.rs

```diff
@@ -122,8 +122,7 @@ pub fn finalize_assets(
       match &mut asset.content {
         StrOrBytes::Str(content) => {
           *content =
-            replace_placeholder_with_hash(mem::take(content), &final_hashes_by_placeholder)
-              .into_owned();
+            replace_placeholder_with_hash(content, &final_hashes_by_placeholder).into_owned();
```

Comment:
Does this cause a extra `clone` if the replacement is not happen?

Reply:
Can you elaborate where is the potential `clone` happen? the function use `&'a str` and  the `Regex::replace_all` return `Cow::borrow` if the there is no matched found in content. ref: 
> Replaces all non-overlapping matches in the haystack with the
replacement provided. This is the same as calling `replacen` with
`limit` set to `0`.

> If no match is found, then the haystack is returned unchanged. In that
case, this implementation will likely return a `Cow::Borrowed` value
such that no allocation is performed.

Reply:
I see